diff --git a/sys/netatalk/endian.h b/sys/netatalk/endian.h
index 5e68c45..3679ba6 100644
--- a/sys/netatalk/endian.h
+++ b/sys/netatalk/endian.h
@@ -1,5 +1,5 @@
 /*
- * $Id: endian.h,v 1.7.12.1 2006/02/08 01:38:42 didg Exp $
+ * $Id: endian.h,v 1.7.12.2 2009/01/10 20:14:29 didg Exp $
  *
  * Copyright (c) 1990,1991 Regents of The University of Michigan.
  * All Rights Reserved. See COPYRIGHT.
@@ -116,7 +116,7 @@ typedef unsigned long long  u_int64_t;
 
 #else /* mips && KERNEL */
 
-#if !( defined( sun ) && defined( i386 ))
+#if !( defined( sun ) && ( defined( i386 ) || defined(__x86_64) ) )
 unsigned short ntohs(), htons();
 unsigned int  ntohl(), htonl();
 #endif /* ! ( sun && i386 ) */
diff --git a/sys/solaris/Makefile.in b/sys/solaris/Makefile.in
index 4127a98..10d3a47 100644
--- a/sys/solaris/Makefile.in
+++ b/sys/solaris/Makefile.in
@@ -1,7 +1,7 @@
 # Solaris specific defines, passed to subdirectories.
 # To use Sun CC, uncomment the CC and KFLAGS variables.
 #
-# $Id: Makefile.in,v 1.1.2.4.2.1 2008/12/20 03:10:54 didg Exp $
+# $Id: Makefile.in,v 1.1.2.4.2.2 2009/01/10 20:14:29 didg Exp $
 # This Makefile.in is not automake generated
 # copy and paste from automake Makefile.in
 srcdir = @srcdir@
@@ -154,5 +154,7 @@ depend :
 
 check:
 
+tags:
+
 # DO NOT DELETE THIS LINE
 
