--- a/distrib/initscripts/Makefile.am
+++ b/distrib/initscripts/Makefile.am
@@ -42,7 +42,7 @@
 	rc.atalk.gentoo.tmpl		\
 	rc.atalk.debian.tmpl
 
-CLEANFILES = $(GENERATED_FILES) atalk afpd atalkd papd timelord
+CLEANFILES = $(GENERATED_FILES) atalk netatalk afpd atalkd papd timelord
 EXTRA_DIST = $(TEMPLATES) rc.atalk.cobalt
 
 # overwrite automake uninstall
@@ -224,18 +224,18 @@
 if USE_DEBIAN
 
 sysvdir = $(DESTDIR)$(sysconfdir)/init.d
-sysv_SCRIPTS = atalk
+sysv_SCRIPTS = netatalk
 
-atalk: rc.atalk.debian
-	cp -f rc.atalk.debian atalk
-	chmod a+x atalk
+netatalk: rc.atalk.debian
+	cp -f rc.atalk.debian netatalk
+	chmod a+x netatalk
 
 install-data-hook:
-	update-rc.d atalk defaults 90 10
+	update-rc.d netatalk defaults 90 10
 
 uninstall-startup:
-	rm -f $(DESTDIR)$(sysconfdir)/init.d/atalk
-	update-rc.d atalk remove
+	rm -f $(DESTDIR)$(sysconfdir)/init.d/netatalk
+	update-rc.d netatalk remove
 
 endif
 
--- a/distrib/initscripts/rc.atalk.debian.tmpl
+++ b/distrib/initscripts/rc.atalk.debian.tmpl
@@ -10,7 +10,7 @@
 
 PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
 DESC="Netatalk"
-NAME=atalk
+NAME=netatalk
 SCRIPTNAME=/etc/init.d/$NAME
 
 # Guard to prevent execution if netatalk was removed.
