--- netatalk/distrib/initscripts/rc.atalk.debian.tmpl.old
+++ netatalk/distrib/initscripts/rc.atalk.debian.tmpl
@@ -16,7 +16,7 @@
 # Guard to prevent execution if netatalk was removed.
 test -x :SBINDIR:/atalkd || exit 0
 
-# Set defaults. Please change these options in :ETCDIR:/netatalk.conf.
+# Set defaults. Please change these options in /etc/default/netatalk.
 AFPD_UAMLIST="-U uams_dhx2.so"
 AFPD_GUEST=nobody
 AFPD_MAX_CLIENTS=50
@@ -28,8 +28,8 @@
 ATALK_UNIX_CHARSET='LOCALE'
 
 # Read in netatalk configuration.
-if [ -f :ETCDIR:/netatalk.conf ]; then
-	. :ETCDIR:/netatalk.conf
+if [ -f /etc/default/netatalk ]; then
+	. /etc/default/netatalk
 fi
 
 # Start Netatalk servers.
