--- netatalk-2.0.3.orig/configure.in
+++ netatalk-2.0.3/configure.in
@@ -697,11 +697,16 @@
 
         # For quotas on Linux XFS filesystems
         
+AC_ARG_WITH([xfs],
+	AS_HELP_STRING([--without-xfs], [Build without XFS filesystem quota support]))
+
+if test "x$with_xfs" != "xno"; then
         # For linux > 2.5.56
         AC_CHECK_HEADERS(linux/dqblk_xfs.h,,
 		[AC_CHECK_HEADERS(linux/xqm.h linux/xfs_fs.h)
         	AC_CHECK_HEADERS(xfs/libxfs.h xfs/xqm.h xfs/xfs_fs.h)]
 	)
+fi
 
 
 	dnl ----- as far as I can tell, dbtob always does the wrong thing
