--- a/configure.in
+++ b/configure.in
@@ -702,13 +702,14 @@
 #		AC_MSG_RESULT([enabling quotactl wrapper])
 #	)
 
-        # For quotas on Linux XFS filesystems
-        
-        # For linux > 2.5.56
-        AC_CHECK_HEADERS(linux/dqblk_xfs.h,,
-		[AC_CHECK_HEADERS(linux/xqm.h linux/xfs_fs.h)
-        	AC_CHECK_HEADERS(xfs/libxfs.h xfs/xqm.h xfs/xfs_fs.h)]
-	)
+dnl	disable for now, code broken: redefines __swabXXX in xfs/swab.h
+dnl	# For quotas on Linux XFS filesystems
+dnl
+dnl	# For linux > 2.5.56
+dnl		AC_CHECK_HEADERS(linux/dqblk_xfs.h,,
+dnl		[AC_CHECK_HEADERS(linux/xqm.h linux/xfs_fs.h)
+dnl		AC_CHECK_HEADERS(xfs/libxfs.h xfs/xqm.h xfs/xfs_fs.h)]
+dnl	)
 
 
 	dnl ----- as far as I can tell, dbtob always does the wrong thing
