--- netatalk.orig/macros/db3-check.m4	2007-12-14 01:26:42.000000000 +0100
+++ netatalk/macros/db3-check.m4	2007-12-14 01:40:56.000000000 +0100
@@ -120,25 +120,36 @@
 AC_DEFUN([NETATALK_BERKELEY_LINK],
 [
 atalk_cv_lib_db=no
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_6,[-ldb-4.6])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db46,[-ldb46])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_46,[-ldb-46])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_6,[-ldb-4-6])
+
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_5,[-ldb-4.5])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db45,[-ldb45])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_45,[-ldb-45])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_5,[-ldb-4-5])
+
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_4,[-ldb-4.4])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db44,[-ldb44])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_44,[-ldb-44])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_4,[-ldb-4-4])
+
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_3,[-ldb-4.3])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db43,[-ldb43])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_43,[-ldb-43])
+NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_3,[-ldb-4-3])
+
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_2,[-ldb-4.2])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db42,[-ldb42])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db_42,[-ldb-42])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_2,[-ldb-4-2])
 
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_2,[-ldb-4.4])
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db42,[-ldb44])
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db_42,[-ldb-44])
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_2,[-ldb-4-4])
-
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_2,[-ldb-4.3])
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db42,[-ldb43])
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db_42,[-ldb-43])
-NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_2,[-ldb-4-3])
-
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_dot_1,[-ldb-4.1])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db41,[-ldb41])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db_41,[-ldb-41])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4_1,[-ldb-4-1])
+
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db_4,[-ldb-4])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db4,[-ldb4])
 NETATALK_BDB_LINK_TRY(atalk_cv_db_db,[-ldb])
@@ -150,7 +161,7 @@
 	trybdbdir=""
 	dobdbsearch=yes
 	bdb_search_dirs="/usr/local/include /usr/include"
-	search_subdirs="/db4.2 /db42 /db4.3 /db43 /db4.4 /db44 /db4.1 /db41 /db4 /"
+	search_subdirs="/db4.6 /db46 /db4.5 /db45 /db4.4 /db44 /db4.3 /db43 /db4.2 /db42 /db4.1 /db41 /db4 /"
 
 dnl required BDB version
 	DB_MAJOR_REQ=4
