--- netatalk/config/AppleVolumes.default.tmpl.old	2005-03-15 12:55:26.970845044 +0100
+++ netatalk/config/AppleVolumes.default.tmpl	2005-03-15 12:57:48.186659354 +0100
@@ -1,7 +1,3 @@
-# This file looks empty when viewed with "vi".  In fact, there is one
-# '~', so users with no AppleVolumes file in their home directory get
-# their home directory by default.
-# 
 # volume format:
 # :DEFAULT: [all of the default options except volume name]
 # path [name] [casefold:x] [options:z,l,j] \
@@ -116,7 +112,5 @@
 # cnidscheme:scheme   -> set the cnid scheme for the volume, default is [:DEFAULT_CNID_SCHEME:]
 #                        available schemes: [:COMPILED_BACKENDS:]
 #
-# The "~" below indicates that Home directories are visible by default.
-# If you do not wish to have people accessing their Home directories,
-# please put a pound sign in front of the tilde or delete it.
-~
+# By default all users have access to their home directories.
+~/			"Home Directory"
