--- a/bin/afile/Makefile.in
+++ b/bin/afile/Makefile.in
@@ -40,7 +40,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-bin_PROGRAMS = afile$(EXEEXT) achfile$(EXEEXT)
+bin_PROGRAMS = apple_file$(EXEEXT) apple_chfile$(EXEEXT)
 subdir = bin/afile
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -76,12 +76,12 @@
 am__installdirs = "$(DESTDIR)$(bindir)" "$(DESTDIR)$(bindir)"
 binPROGRAMS_INSTALL = $(INSTALL_PROGRAM)
 PROGRAMS = $(bin_PROGRAMS)
-am_achfile_OBJECTS = achfile.$(OBJEXT) common.$(OBJEXT)
-achfile_OBJECTS = $(am_achfile_OBJECTS)
-achfile_LDADD = $(LDADD)
-am_afile_OBJECTS = afile.$(OBJEXT) common.$(OBJEXT)
-afile_OBJECTS = $(am_afile_OBJECTS)
-afile_LDADD = $(LDADD)
+am_apple_chfile_OBJECTS = achfile.$(OBJEXT) common.$(OBJEXT)
+apple_chfile_OBJECTS = $(am_apple_chfile_OBJECTS)
+apple_chfile_LDADD = $(LDADD)
+am_apple_file_OBJECTS = afile.$(OBJEXT) common.$(OBJEXT)
+apple_file_OBJECTS = $(am_apple_file_OBJECTS)
+apple_file_LDADD = $(LDADD)
 binSCRIPT_INSTALL = $(INSTALL_SCRIPT)
 SCRIPTS = $(bin_SCRIPTS)
 DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(top_builddir)
@@ -95,8 +95,8 @@
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
-SOURCES = $(achfile_SOURCES) $(afile_SOURCES)
-DIST_SOURCES = $(achfile_SOURCES) $(afile_SOURCES)
+SOURCES = $(apple_chfile_SOURCES) $(apple_file_SOURCES)
+DIST_SOURCES = $(apple_chfile_SOURCES) $(apple_file_SOURCES)
 ETAGS = etags
 CTAGS = ctags
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
@@ -311,8 +311,8 @@
 target_vendor = @target_vendor@
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/sys
 bin_SCRIPTS = acleandir.rc
-afile_SOURCES = afile.c common.c common.h
-achfile_SOURCES = achfile.c common.c common.h
+apple_file_SOURCES = afile.c common.c common.h
+apple_chfile_SOURCES = achfile.c common.c common.h
 EXTRA_DIST = acleandir.rc
 all: all-am
 
@@ -375,12 +375,12 @@
 	  echo " rm -f $$p $$f"; \
 	  rm -f $$p $$f ; \
 	done
-achfile$(EXEEXT): $(achfile_OBJECTS) $(achfile_DEPENDENCIES) 
-	@rm -f achfile$(EXEEXT)
-	$(LINK) $(achfile_LDFLAGS) $(achfile_OBJECTS) $(achfile_LDADD) $(LIBS)
-afile$(EXEEXT): $(afile_OBJECTS) $(afile_DEPENDENCIES) 
-	@rm -f afile$(EXEEXT)
-	$(LINK) $(afile_LDFLAGS) $(afile_OBJECTS) $(afile_LDADD) $(LIBS)
+apple_chfile$(EXEEXT): $(apple_chfile_OBJECTS) $(apple_chfile_DEPENDENCIES) 
+	@rm -f apple_chfile$(EXEEXT)
+	$(LINK) $(apple_chfile_LDFLAGS) $(apple_chfile_OBJECTS) $(apple_chfile_LDADD) $(LIBS)
+apple_file$(EXEEXT): $(apple_file_OBJECTS) $(apple_file_DEPENDENCIES) 
+	@rm -f apple_file$(EXEEXT)
+	$(LINK) $(apple_file_LDFLAGS) $(apple_file_OBJECTS) $(apple_file_LDADD) $(LIBS)
 install-binSCRIPTS: $(bin_SCRIPTS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(mkdir_p) "$(DESTDIR)$(bindir)"
