--- a/distrib/initscripts/Makefile.in
+++ b/distrib/initscripts/Makefile.in
@@ -314,7 +314,7 @@
 	rc.atalk.gentoo.tmpl		\
 	rc.atalk.debian.tmpl
 
-CLEANFILES = $(GENERATED_FILES) atalk afpd atalkd papd timelord
+CLEANFILES = $(GENERATED_FILES) atalk netatalk afpd atalkd papd timelord
 EXTRA_DIST = $(TEMPLATES) rc.atalk.cobalt
 
 #
@@ -357,7 +357,7 @@
 #
 @USE_TRU64_TRUE@sysvdir = /sbin/init.d
 @USE_COBALT_TRUE@sysv_SCRIPTS = atalk
-@USE_DEBIAN_TRUE@sysv_SCRIPTS = atalk
+@USE_DEBIAN_TRUE@sysv_SCRIPTS = netatalk
 @USE_GENTOO_TRUE@sysv_SCRIPTS = atalk
 @USE_NETBSD_TRUE@sysv_SCRIPTS = afpd atalkd papd timelord
 @USE_REDHAT_TRUE@sysv_SCRIPTS = atalk
@@ -658,16 +658,16 @@
 @USE_GENTOO_TRUE@	-rc-update del atalk default
 @USE_GENTOO_TRUE@	rm -f /etc/init.d/atalk
 
-@USE_DEBIAN_TRUE@atalk: rc.atalk.debian
-@USE_DEBIAN_TRUE@	cp -f rc.atalk.debian atalk
-@USE_DEBIAN_TRUE@	chmod a+x atalk
+@USE_DEBIAN_TRUE@netatalk: rc.atalk.debian
+@USE_DEBIAN_TRUE@	cp -f rc.atalk.debian netatalk
+@USE_DEBIAN_TRUE@	chmod a+x netatalk
 
 @USE_DEBIAN_TRUE@install-data-hook:
-@USE_DEBIAN_TRUE@	update-rc.d atalk defaults 90 10
+@USE_DEBIAN_TRUE@	update-rc.d netatalk defaults 90 10
 
 @USE_DEBIAN_TRUE@uninstall-startup:
-@USE_DEBIAN_TRUE@	rm -f /etc/init.d/atalk
-@USE_DEBIAN_TRUE@	update-rc.d atalk remove
+@USE_DEBIAN_TRUE@	rm -f /etc/init.d/netatalk
+@USE_DEBIAN_TRUE@	update-rc.d netatalk remove
 
 #
 # defaults, no init scripts installed
