--- netatalk/distrib/initscripts/Makefile.in.old
+++ netatalk/distrib/initscripts/Makefile.in
@@ -584,11 +584,11 @@
 @USE_DEBIAN_TRUE@	chmod a+x atalk
 
 @USE_DEBIAN_TRUE@install-data-hook:
-@USE_DEBIAN_TRUE@	update-rc.d netatalk defaults 90 10
+@USE_DEBIAN_TRUE@#	update-rc.d netatalk defaults 90 10
 
 @USE_DEBIAN_TRUE@uninstall-startup:
-@USE_DEBIAN_TRUE@	rm -f /etc/init.d/netatalk
-@USE_DEBIAN_TRUE@	update-rc.d netatalk remove
+@USE_DEBIAN_TRUE@#	rm -f /etc/init.d/netatalk
+@USE_DEBIAN_TRUE@#	update-rc.d netatalk remove
 
 #
 # defaults, no init scripts installed
