--- a/config/netatalk.conf
+++ b/config/netatalk.conf
@@ -28,7 +28,7 @@
 PAPD_RUN=no
 TIMELORD_RUN=no
 A2BOOT_RUN=no
-CNID_METAD_RUN=yes
+CNID_METAD_RUN=no
 AFPD_RUN=yes
 
 # Control whether the daemons are started in the background.
--- a/distrib/initscripts/rc.atalk.debian.tmpl
+++ b/distrib/initscripts/rc.atalk.debian.tmpl
@@ -28,7 +28,7 @@
 ATALK_ZONE=
 ATALK_NAME=`/bin/hostname --short`
 ATALK_BGROUND=no
-CNID_METAD_RUN=yes
+CNID_METAD_RUN=no
 ATALK_MAC_CHARSET='MAC_ROMAN'
 ATALK_UNIX_CHARSET='LOCALE'
 
