--- a/config/Makefile.in
+++ b/config/Makefile.in
@@ -624,7 +624,7 @@
 			echo "not overwriting $$f"; \
 		fi; \
 	done
-@USE_DEBIAN_TRUE@	$(mkinstalldirs) $(DESTDIR)/default
+@USE_DEBIAN_TRUE@	$(mkinstalldirs) $(DESTDIR)/etc/default
 @USE_DEBIAN_TRUE@	if test "x$(OVERWRITE_CONFIG)" = "xyes" -o ! -f /etc/default/netatalk; then \
 @USE_DEBIAN_TRUE@		echo "$(INSTALL_DATA) $(srcdir)/netatalk.conf $(DESTDIR)/etc/default/netatalk"; \
 @USE_DEBIAN_TRUE@		$(INSTALL_DATA) $(srcdir)/netatalk.conf $(DESTDIR)/etc/default/netatalk; \
