diff -ruN netatalk-3.0.3.orig/distrib/initscripts/rc.debian.tmpl netatalk-3.0.3/distrib/initscripts/rc.debian.tmpl
--- netatalk-3.0.3.orig/distrib/initscripts/rc.debian.tmpl	2013-01-04 15:37:06.000000000 +0100
+++ netatalk-3.0.3/distrib/initscripts/rc.debian.tmpl	2013-04-11 13:08:20.375689666 +0200
@@ -5,7 +5,7 @@
 # Should-Start:      avahi-daemon
 # Required-Stop:     $remote_fs $syslog
 # Default-Start:     2 3 4 5
-# Default-Stop:      1
+# Default-Stop:      0 1 6
 ### END INIT INFO
 #
 # netatalk      Netatalk :NETATALK_VERSION: initscript
